.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: Test API
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - run: docker --version
+      - uses: actions/checkout@v4
+      - name: Build Docker image
+        run: docker build --tag dracor/metrics .
```

Comment:
Yes, I thought about adding unit tests, that run without the Docker container. Of course that is faster.

But I went for testing the Docker container, because:

- I guess this project is never used without being in a container
- this way we can test, changing the base image (like Python 3.9, 3.10, or python, python-slim etc) and still getting the same result
- I would like to propose to leave pipenv and switch to another package manager later
- I would like to propose to leave hug (not working with python 3.12, and the last release is from 2019)

What do you think?

Reply:
I would opt for unit tests outside of docker. This workflow is supposed to run for each push to `main` or pull request. Few of those will actually change the Dockerfile. And if they do I would trust the developer or at least the reviewer of a PR to have tested the resulting docker container. Your pytest should make that easy enough now. With that in mind I think we can save some cycles here. 

> * I guess this project is never used without being in a container

On the dracor.org VM the metrics service actually does not run in a container. The Dockerfile so far is mainly used to facilitate a local Dracor instance.

> * this way we can test, changing the base image (like Python 3.9, 3.10, or python, python-slim etc) and still getting the same result

See above. Need not be done on each and every code change, developer diligence is good enough for now.

> * I would like to propose to leave pipenv and switch to another package manager later
> * I would like to propose to leave hug (not working with python 3.12, and the last release is from 2019)

That maybe something to consider but is off-topic here and should be dealt with in separate issues. Feel free to open them ;)

Reply:
Thanks. This repo rarely changes so nobody felt the need for testing it so far. But it's nice to see someone has an eye on this now :)